tests/test_logic.py

```diff
@@ -2524,7 +2524,7 @@ def test_change_mode_workspace_dir_exception():
         ed.change_mode("circuitpython")
         assert mock_error.call_count == 1
     assert ed.mode == "circuitpython"
-    assert python_mode.workspace_dir.called_once()
+    python_mode.workspace_dir.assert_called_once()
```

Comment:
Looks like `assert_called_once()` was introduced in Python 3.6, so it fails in 3.5.
We are about to update dependencies and drop support for all the legacy stuff, but in the meantime I'll update this to use assert_called_once_with() and it should pass tests.